client/extensions/woocommerce/state/data-layer/action-list/index.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * Internal dependencies
+ */
+import { getActionList } from 'woocommerce/state/action-list/selectors';
+import { actionListStepStart } from 'woocommerce/state/action-list/actions';
+import {
+	WOOCOMMERCE_ACTION_LIST_STEP_START,
+	WOOCOMMERCE_ACTION_LIST_STEP_SUCCESS,
+} from 'woocommerce/state/action-types';
+
+export function handleStepStart( { dispatch, getState }, action ) {
+	const { stepIndex } = action;
+
+	const actionList = getActionList( getState() );
+	const step = actionList[ stepIndex ];
+	dispatch( step.action );
+}
+
+export function handleStepSuccess( { dispatch, getState }, action ) {
+	const { stepIndex, time } = action;
+
+	// TODO: Consider a bit of sanity checking to ensure the next index is the right one?
+
+	const actionList = getActionList( getState() );
+	const nextStepIndex = stepIndex + 1;
```

Comment:
Yeah, I think it made more sense to do it this way when it was just chaining actions, before it became a data-layer handler. There's really no reason to not use selectors when you always have access to the state.